### PR TITLE
Added feature policy tests for header feature policy

### PR DESCRIPTION
**Pull request by LoonyBean**
Created a directory for feature policy tests. This pr include tests for feature policy defined by http header (test fullscreen, payment, and vibrate). More tests for iframe allow attribute will be included in the next pr.

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/208342072)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 7cc179d7e8bcd807724b08bba4549c84124c2d20
Using browser at version BuildID 20170306110339; SourceStamp 966464a68a2cb3ca1125808e34abb5c1d34e3797
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>9 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5055/feature-policy/header-policy-fullscreen-disabled.sub.html">/feature-policy/header-policy-fullscreen-disabled.sub.html</a></summary>


|                                                           Subtest                                                            | Results |     Messages     |
|------------------------------------------------------------------------------------------------------------------------------|---------|------------------|
|                                                                                                                              | TIMEOUT |                  |
| `same origin iframe with allowfullscreen = false may not call document.requestFullscreen() when disabled by feature policy`  | TIMEOUT | `Test timed out` |
| `cross origin iframe with allowfullscreen = false may not call document.requestFullscreen() when disabled by feature policy` | NOTRUN  |                  |
| `same origin iframe with allowfullscreen = true may not call document.requestFullscreen() when disabled by feature policy`   | NOTRUN  |                  |
| `cross origin iframe with allowfullscreen = true may not call document.requestFullscreen() when disabled by feature policy`  | NOTRUN  |                  |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5055/feature-policy/header-policy-fullscreen-enabled-for-all.sub.html">/feature-policy/header-policy-fullscreen-enabled-for-all.sub.html</a></summary>


|                                                         Subtest                                                         | Results |     Messages     |
|-------------------------------------------------------------------------------------------------------------------------|---------|------------------|
|                                                                                                                         | TIMEOUT |                  |
| `same origin iframe with allowfullscreen = false may call document.requestFullscreen() when enabled by feature policy`  | TIMEOUT | `Test timed out` |
| `cross origin iframe with allowfullscreen = false may call document.requestFullscreen() when enabled by feature policy` | NOTRUN  |                  |
| `same origin iframe with allowfullscreen = true may call document.requestFullscreen() when enabled by feature policy`   | NOTRUN  |                  |
| `cross origin iframe with allowfullscreen = true may call document.requestFullscreen() when enabled by feature policy`  | NOTRUN  |                  |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5055/feature-policy/header-policy-fullscreen-enabled-for-self.sub.html">/feature-policy/header-policy-fullscreen-enabled-for-self.sub.html</a></summary>


|                                                                       Subtest                                                                       | Results |     Messages     |
|-----------------------------------------------------------------------------------------------------------------------------------------------------|---------|------------------|
|                                                                                                                                                     | TIMEOUT |                  |
| `same origin iframe with allowfullscreen = false may call document.requestFullscreen() when enabled by feature policy or allowfullscreen attribute` | TIMEOUT | `Test timed out` |
| `cross origin iframe with allowfullscreen = false may not call document.requestFullscreen() when disabled by feature policy`                        | NOTRUN  |                  |
| `same origin iframe with allowfullscreen = true may call document.requestFullscreen() when enabled by feature policy or allowfullscreen attribute`  | NOTRUN  |                  |
| `cross origin iframe with allowfullscreen = true may not call document.requestFullscreen() when disabled by feature policy`                         | NOTRUN  |                  |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5055/feature-policy/header-policy-payment-disabled.sub.html">/feature-policy/header-policy-payment-disabled.sub.html</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | TIMEOUT |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5055/feature-policy/header-policy-payment-enabled-for-all.sub.html">/feature-policy/header-policy-payment-enabled-for-all.sub.html</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | TIMEOUT |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5055/feature-policy/header-policy-payment-enabled-for-self.sub.html">/feature-policy/header-policy-payment-enabled-for-self.sub.html</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | TIMEOUT |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5055/feature-policy/header-policy-vibrate-disabled.sub.html">/feature-policy/header-policy-vibrate-disabled.sub.html</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | TIMEOUT |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5055/feature-policy/header-policy-vibrate-enabled-for-all.sub.html">/feature-policy/header-policy-vibrate-enabled-for-all.sub.html</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | TIMEOUT |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5055/feature-policy/header-policy-vibrate-enabled-for-self.sub.html">/feature-policy/header-policy-vibrate-enabled-for-self.sub.html</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | TIMEOUT |          |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/208342074)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 7cc179d7e8bcd807724b08bba4549c84124c2d20
Using browser at version 58.0.3026.3 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>9 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5055/feature-policy/header-policy-fullscreen-disabled.sub.html">/feature-policy/header-policy-fullscreen-disabled.sub.html</a></summary>


|                                                           Subtest                                                            | Results |     Messages     |
|------------------------------------------------------------------------------------------------------------------------------|---------|------------------|
|                                                                                                                              | TIMEOUT |                  |
| `same origin iframe with allowfullscreen = false may not call document.requestFullscreen() when disabled by feature policy`  | TIMEOUT | `Test timed out` |
| `cross origin iframe with allowfullscreen = false may not call document.requestFullscreen() when disabled by feature policy` | NOTRUN  |                  |
| `same origin iframe with allowfullscreen = true may not call document.requestFullscreen() when disabled by feature policy`   | NOTRUN  |                  |
| `cross origin iframe with allowfullscreen = true may not call document.requestFullscreen() when disabled by feature policy`  | NOTRUN  |                  |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5055/feature-policy/header-policy-fullscreen-enabled-for-all.sub.html">/feature-policy/header-policy-fullscreen-enabled-for-all.sub.html</a></summary>


|                                                         Subtest                                                         | Results |     Messages     |
|-------------------------------------------------------------------------------------------------------------------------|---------|------------------|
|                                                                                                                         | TIMEOUT |                  |
| `same origin iframe with allowfullscreen = false may call document.requestFullscreen() when enabled by feature policy`  | TIMEOUT | `Test timed out` |
| `cross origin iframe with allowfullscreen = false may call document.requestFullscreen() when enabled by feature policy` | NOTRUN  |                  |
| `same origin iframe with allowfullscreen = true may call document.requestFullscreen() when enabled by feature policy`   | NOTRUN  |                  |
| `cross origin iframe with allowfullscreen = true may call document.requestFullscreen() when enabled by feature policy`  | NOTRUN  |                  |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5055/feature-policy/header-policy-fullscreen-enabled-for-self.sub.html">/feature-policy/header-policy-fullscreen-enabled-for-self.sub.html</a></summary>


|                                                                       Subtest                                                                       | Results |     Messages     |
|-----------------------------------------------------------------------------------------------------------------------------------------------------|---------|------------------|
|                                                                                                                                                     | TIMEOUT |                  |
| `same origin iframe with allowfullscreen = false may call document.requestFullscreen() when enabled by feature policy or allowfullscreen attribute` | TIMEOUT | `Test timed out` |
| `cross origin iframe with allowfullscreen = false may not call document.requestFullscreen() when disabled by feature policy`                        | NOTRUN  |                  |
| `same origin iframe with allowfullscreen = true may call document.requestFullscreen() when enabled by feature policy or allowfullscreen attribute`  | NOTRUN  |                  |
| `cross origin iframe with allowfullscreen = true may not call document.requestFullscreen() when disabled by feature policy`                         | NOTRUN  |                  |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5055/feature-policy/header-policy-payment-disabled.sub.html">/feature-policy/header-policy-payment-disabled.sub.html</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | TIMEOUT |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5055/feature-policy/header-policy-payment-enabled-for-all.sub.html">/feature-policy/header-policy-payment-enabled-for-all.sub.html</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | TIMEOUT |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5055/feature-policy/header-policy-payment-enabled-for-self.sub.html">/feature-policy/header-policy-payment-enabled-for-self.sub.html</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | TIMEOUT |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5055/feature-policy/header-policy-vibrate-disabled.sub.html">/feature-policy/header-policy-vibrate-disabled.sub.html</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | TIMEOUT |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5055/feature-policy/header-policy-vibrate-enabled-for-all.sub.html">/feature-policy/header-policy-vibrate-enabled-for-all.sub.html</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | TIMEOUT |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5055/feature-policy/header-policy-vibrate-enabled-for-self.sub.html">/feature-policy/header-policy-vibrate-enabled-for-self.sub.html</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | TIMEOUT |          |

</details>
</details>

File: feature-policy/OWNERS [hunk 1]
```diff
@@ -0,0 +1,2 @@
+@iclelland
```
Comment (foolip) [inline feature-policy/OWNERS]: These should be the GitHub user names.

Comment (clelland) [inline feature-policy/OWNERS]: `@clelland` and `@LoonyBean`, in this case

File: feature-policy/header-policy-payment-disabled.sub.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+<script src="/resources/testharnessreport.js"></script>
+<script>
+  if (window.testRunner) {
+    testRunner.dumpAsText();
+    testRunner.waitUntilDone();
+  }
+</script>
+<iframe id="same-origin" src="resources/vibrate-disabled.html"></iframe>
+<iframe id="cross-origin" src="{{location[scheme]}}://{{domains[www]}}:{{location[port]}}/feature-policy/resources/vibrate-disabled.html"></iframe>
+<iframe id="same-origin" src="resources/vibrate-disabled.html"></iframe>
```
Comment (clelland) [inline feature-policy/header-policy-payment-disabled.sub.html]: These two frames should have different ids from the ones in line 10 and 11 -- It's an error to have two elements in the same document with the same id attribute
(https://html.spec.whatwg.org/#the-id-attribute)

Comment (clelland) [inline feature-policy/header-policy-payment-disabled.sub.html]: (Or else they should be deleted -- they look identical to the other two lines, so I'm not sure if they're there for a reason, or if it is accidental.)

File: feature-policy/header-policy-fullscreen-disabled-manual.sub.html [hunk 3]
```diff
@@ -0,0 +1,41 @@
+<!DOCTYPE html>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+  if (window.testRunner) {
```
Comment (clelland) [inline feature-policy/header-policy-fullscreen-disabled-manual.sub.html]: We use this in Chrome's layout test runner, but do you know if it has any actual function inside of web platform tests? (I don't see much usage of it in this repo, except in a blink-imported test)

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5055/)

**Comment by LoonyBean** [comment]
Changes are made, PTAL

**Comment by foolip** [comment]
@LoonyBean given that https://github.com/w3c/web-platform-tests/tree/master/feature-policy now exists I guess you did this on another PR or via Chromium export instead? Should we close this PR?

**Comment by loonybear** [comment]
Yea I think we should just close this. Thanks @foolip

**Pull request closed without merging**